### DISCUSSION TITLE
[Feature Request] Provide an easy way to populate all the video URLs from a YouTube playlist into the syncplay playlist

### DISCUSSION THREAD

**Discussion by jvschiavo** [Syncplay/syncplay#449] [Archive]
Let's say I have some generic Youtube playlist like this one: https://www.youtube.com/playlist?list=PL2788304DC59DBEB4

![image](https://user-images.githubusercontent.com/20977075/112620844-cc64b780-8e07-11eb-8532-058129cb96b7.png)

If I add this URL to syncplay, although it makes it possible to watch all the videos by checking Next on the video player, it doesn't add all the video links to the syncplay playlist, making it harder to choose a specific video to play for everyone in the room.

Is there an easy way to populate all the video URLs from the YouTube playlist into the syncplay playlist?

![image](https://user-images.githubusercontent.com/20977075/112621864-03879880-8e09-11eb-855a-f49307b721ca.png)
This could be a possible intended behavior.

**Comment by Et0h** [comment]
Syncplay doesn't have any easy feature to do this, but you might be able to use the `-j --flat-playlist` feature of youtude-dl and then parse the results.

**Comment by jvschiavo** [comment]
> Syncplay doesn't have any easy feature to do this, but you might be able to use the `-j --flat-playlist` feature of youtude-dl and then parse the results.

Yaaay, it helped!
Thanks to you and a friend of mine I was able to write a function that does exactly this! :)

It looks like this:
![yt-list function](https://user-images.githubusercontent.com/20977075/113525732-02830380-958d-11eb-9ed8-06d52f8266b7.png)

Could we maybe implement something like this directly into syncplay? When the user pastes the youtube playlist link it could maybe create a collapsible drop-down with all the videos of the playlist listed, so anyone could open the dropdown and choose a specific video to play.

Edit: the code used for the function:

```
yt-list()
{
  youtube-dl -j --flat-playlist "$1" | sed 's/.*\"url\": \"\([^\"]*\)\".*/https:\/\/www.youtube.com\/watch?v=\1/'
}
```

**Comment by Et0h** [comment]
I'm not sure how easy it would be to directly integrate the feature into Syncplay on all platforms and whether on balance we want the added complexity or not, but I can see why some people would find it useful. 

My initial expectation would be that Syncplay would run youtube-dl (if the user has it) with those options and parse the results rather than try to make and maintain its own system for decoding YouTube playlists. This would require the ytdl location to be specified somewhere.

**Comment by DuendeInexistente** [comment]
> I'm not sure how easy it would be to directly integrate the feature into Syncplay on all platforms and whether on balance we want the added complexity or not, but I can see why some people would find it useful.
> 
> My initial expectation would be that Syncplay would run youtube-dl (if the user has it) with those options and parse the results rather than try to make and maintain its own system for decoding YouTube playlists. This would require the ytdl location to be specified somewhere.

Why not just use %path% or a local folder?
If syncplay does use youtube-dl and pack its own copy, you'd probby want to also run youtube-dl -U automatically on each launch for users who don't know how to update it themselves.

**Comment by Et0h** [comment]
This discussion is being locked as there are no plans to add this feature to Syncplay. Adding non-core features not only takes up development time but can increase the complexity of the software, increasing the chance of bugs and making the interface and documentation more complex. As per #315 we only have limited development time, and so we are focussing on supporting the main use cases and not every possible way that people might want to use Syncplay.
